requirements-types.txt

```diff
@@ -0,0 +1,2 @@
+--requirement requirements.txt
+--requirement reqs/requirements-dev-contrib.txt
```

Comment:
I would put this in `/reqs` but I need to be able to install the core `gx` requirements.

I tried just putting `.` which should have just installed the python package in the CWD but I got a `pip` error. (I thought this used to allowed 🤔 ).

I will see if I can do something like...
```
--requirements ../requirements.txt
```

Reply:
Ok that actually worked 😄

Reply:
It "worked" but reverted because it failed on `import ge` step. 
`setup.py` `get_extras_require()` could not parse the `../requirements.txt` line.
Didn't seem worth fixing.

Reply:
In the pipeline you could just do two steps, `pip install -r requirements.txt` and then `pip install reqs/requirements-types.txt`. That would be more consistent I think.

Reply:
I could but I would prefer this as requiring two different install commands makes it easier for the different type-checking checks to get out of sync again.

Reply:
If this requirements-types.txt file is just going to be the base package requirements and those from contrib, I think it makes sense to allow contrib to be an extras_requires key (instead of popping it off) and then do `pip install ".[contrib]"

Just curious, why is the full requirements.txt needed in pipeline stages where doing `pip install --requirement requirements-types.txt`?

Reply:
>Just curious, why is the full requirements.txt needed in pipeline stages where doing pip install --requirement requirements-types.txt?

The typing checker needs those dependencies to do its type-checking.
For example in order to know whether `sqlalchemy` types and methods are being used correctly, it needs `sqlalchemy` installed.

Reply:
>If this requirements-types.txt file is just going to be the base package requirements and those from contrib, I think it makes sense to allow contrib to be an extras_requires key (instead of popping it off) and then do `pip install ".[contrib]"

Ah okay, so we could have the static type-checking step be...
```
pip install .[contrib]. # or .[types] or something
invoke --ci
```
And that would always work unless something was broken in the package install.
Although it would be slower, and if the package install does break, the type-checker could actually be helpful there 🤔